ui/app/src/views/ViewDashboard.tsx

```diff
@@ -52,7 +52,12 @@ function ViewDashboard() {
 
   const navigate = useNavigate();
   const { successSnackbar, exceptionSnackbar, warningSnackbar } = useSnackbar();
-  const datasourceApi = useDatasourceApi();
+  const [datasourceApi] = useState(new CachedDatasourceAPI(new HTTPDatasourceAPI()));
+  useEffect(() => {
+    // warm up the caching of the datasources
+    datasourceApi.listDatasources(projectName, undefined);
```

Comment:
If pluginKind is optional, would just not passing anything work instead of setting `undefined`?

Reply:
yeah you are totally right. I think the auto implementation of the `DatasourceAPI` of IntelliJ messed up.